script.module.xbmcswift2/lib/xbmcswift2/listitem.py

```diff
@@ -33,36 +31,41 @@ def __init__(self, label=None, label2=None, icon=None, thumbnail=None,
                       if val is not None)
         self._listitem = xbmcgui.ListItem(**kwargs)
 
-        # xbmc doesn't make getters available for these properties so we'll
+        # kodi doesn't make getters available for these properties so we'll
         # keep track on our own
-        self._icon = icon
         self._path = path
-        self._thumbnail = thumbnail
         self._context_menu_items = []
         self.is_folder = True
         self._played = False
+        self._art = {'icon': icon, 'thumb': thumbnail, 'fanart': fanart}
+
+        # set listitem art
+        self._listitem.setArt(self._art)
 
     def __repr__(self):
+        if PY3:
+            return ("<ListItem '%s'>" % self.label)
         return ("<ListItem '%s'>" % self.label).encode('utf-8')
 
     def __str__(self):
+        if PY3:
+            return ('%s (%s)' % (self.label, self.path))
         return ('%s (%s)' % (self.label, self.path)).encode('utf-8')
 
     def get_context_menu_items(self):
         '''Returns the list of currently set context_menu items.'''
         return self._context_menu_items
 
-    def add_context_menu_items(self, items, replace_items=False):
```

Comment:
This is a breaking change and now function documentation is also wrong.

Reply:
Will adjust the docs